app/org/sagebionetworks/bridge/upload/UploadValidationTask.java

```diff
@@ -111,6 +124,65 @@ public void run() {
         }
 
         // TODO: if validation fails, wipe the files from S3
+
+        // dedupe logic over here:
+        try {
+            dedupeHelper(context.getRecordId());
+        } catch (Throwable e) {
+            logErrorMsg(e);
+        }
+
+    }
+
+    // helper method to log exception
+    void logErrorMsg(Throwable e) {
+        logger.error("An error occurred:", e);
+    }
+
+    /**
+     * helper method: query healthdatarecord by healthcode, schema and createdOn from the upload just finished above,
+     * if there are more than 1 such records in ddb, there are duplicates. log that information but not stop or delete anything right now
+     * @param healthRecordId
+     */
+    private void dedupeHelper(String healthRecordId) throws BadRequestException {
+        if (healthRecordId == null) {
+            return; // no record yet, no need to check duplicates
+        }
+        // get necessary information for query first
+        HealthDataRecord record = healthDataService.getRecordById(healthRecordId);
+
+        // if there is no record in ddb, we should just finish dedupe logic since no record is a common case
+        if (record == null) {
+            return;
+        }
+
+        Long createdOn = record.getCreatedOn();
+        String healthCode = record.getHealthCode();
+        String schemaId = record.getSchemaId();
+
+        List<HealthDataRecord> retList = healthDataService.getRecordsByHealthcodeCreatedOnSchemaId(healthCode, createdOn, schemaId);
+
+        if (retList.size() > 1) {
+            logger.info(String.format("Duplicate health data records for record id: %s, created on:  %s, schema id: %s, duplicate size: %s",
+                    healthRecordId, createdOn, schemaId, retList.size()));
+
+            logDuplicateUploadRecords(record, retList);
+        }
+        // else there is only one such record exists in ddb or no record exists yet, means no duplicate -- do nothing
+    }
+
+    void logDuplicateUploadRecords(HealthDataRecord originRecord, List<HealthDataRecord> dupeRecords) {
+        logger.info("Origin Record: " + "record id: " + originRecord.getId());
+
+        for (HealthDataRecord dupeRecord: dupeRecords) {
+            if (!dupeRecord.getId().equals(originRecord.getId())) {
+                logger.info("Duplicate HealthDataRecord: " + "record id: " + dupeRecord.getId()
+                        + ", createdOn: " + dupeRecord.getCreatedOn().toString()
+                        + ", schemaId: " + dupeRecord.getSchemaId()
+                        + ", studyId: " + dupeRecord.getStudyId() + ", uploadDate: " + dupeRecord.getUploadDate().toString()
+                        + ", uploadId: " + dupeRecord.getUploadId());
+            }
```

Comment:
We want to log the original record ID and the dupe record ID in the same log line. Otherwise, it'll be difficult to match up what record is a duplicate of what record.